[SPARK-27445][SQL][TEST] Update SQLQueryTestSuite to process files ending with `.sql`

dilipbiswal opened the pull request:
## What changes were proposed in this pull request?
While using vi or vim to edit the test files the .swp or .swo files are created and attempt to run the test suite in the presence of these files causes errors like below :
```
nfo] - subquery/exists-subquery/.exists-basic.sql.swp *** FAILED *** (117 milliseconds)
[info]   java.io.FileNotFoundException: /Users/dbiswal/mygit/apache/spark/sql/core/target/scala-2.12/test-classes/sql-tests/results/subquery/exists-subquery/.exists-basic.sql.swp.out (No such file or directory)
[info]   at java.io.FileInputStream.open0(Native Method)
[info]   at java.io.FileInputStream.open(FileInputStream.java:195)
[info]   at java.io.FileInputStream.<init>(FileInputStream.java:138)
[info]   at org.apache.spark.sql.catalyst.util.package$.fileToString(package.scala:49)
[info]   at org.apache.spark.sql.SQLQueryTestSuite.runQueries(SQLQueryTestSuite.scala:247)
[info]   at org.apache.spark.sql.SQLQueryTestSuite.$anonfun$runTest$11(SQLQueryTestSuite.scala:192)
```
~~This minor pr adds these temp files in the ignore list.~~
While computing the list of test files to process, only consider files with `.sql` extension. This makes sure the unwanted temp files created from various editors are ignored from processing.
## How was this patch tested?
Verified manually.

dilipbiswal committed:
In SQLQueryTestSuite ignore .swp and .swo files from processinge

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9797/
Test PASSed.

SparkQA commented:
**[Test build #104467 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104467/testReport)** for PR 24333 at commit [`7fc82d7`](https://github.com/apache/spark/commit/7fc82d7262ce7684ccf45f657dd1cde3d95426ae).

SparkQA commented:
**[Test build #104467 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104467/testReport)** for PR 24333 at commit [`7fc82d7`](https://github.com/apache/spark/commit/7fc82d7262ce7684ccf45f657dd1cde3d95426ae).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/104467/
Test FAILed.

dilipbiswal commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9802/
Test PASSed.

SparkQA commented:
**[Test build #104472 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104472/testReport)** for PR 24333 at commit [`7fc82d7`](https://github.com/apache/spark/commit/7fc82d7262ce7684ccf45f657dd1cde3d95426ae).

SparkQA commented:
**[Test build #104472 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104472/testReport)** for PR 24333 at commit [`7fc82d7`](https://github.com/apache/spark/commit/7fc82d7262ce7684ccf45f657dd1cde3d95426ae).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/104472/
Test PASSed.

attilapiros reviewed (commented):
I am happy somebody thinks about vim users (like me) and the change is fine but I think a solution which combines backlisting with whitelisting would be better: like every file with ".sql" extension and not backlisted. 

This way much more editors which creates temporary files here could be safe to be used.
Even would be better for Vim itself:

> The more .%.swX files are created whenever there are .%.swp and .%.swo and .%.swY files, where X and Y are characters defined in order by Vim.

And this condition is already satisfied:
```
$ find  sql/core/src/test/resources/sql-tests/inputs/ -type f | wc -l
     107
$ find  sql/core/src/test/resources/sql-tests/inputs/ -type f -name "*.sql" | wc -l
     107
```

What is your opinion?

srowen commented:
These files aren't in the repo, and a .swp file only exists if you have the file open in vim. I don't think we should work around it for this reason. However, white-listing .sql files seems cleaner anyway so I'm OK with that.

dilipbiswal committed:
fix1

dilipbiswal commented:
@attilapiros @srowen White-listing .sql files sounds good to me :-)

SparkQA commented:
**[Test build #104492 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104492/testReport)** for PR 24333 at commit [`4c0f1bd`](https://github.com/apache/spark/commit/4c0f1bd61339188755a90d517cea143d67564810).

dilipbiswal committed:
fix1

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9822/
Test PASSed.

SparkQA commented:
**[Test build #104493 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104493/testReport)** for PR 24333 at commit [`eea0a25`](https://github.com/apache/spark/commit/eea0a250f20edb0faadd7ae9c0ee8fcf6098e8d1).

dilipbiswal committed:
remove ds_store from blacklist

SparkQA commented:
**[Test build #104494 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104494/testReport)** for PR 24333 at commit [`78442e3`](https://github.com/apache/spark/commit/78442e3dea8d32e93ad6b25f79df0f66f1a1e7f1).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9823/
Test PASSed.

srowen commented on sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala:
```diff
@@ -329,7 +329,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   private def listFilesRecursively(path: File): Seq[File] = {
     val (dirs, files) = path.listFiles().partition(_.isDirectory)
-    files ++ dirs.flatMap(listFilesRecursively)
+    val filteredFiles = files.filter (_.getName matches validTestFileExtensionsRegEx)
```
Let's avoid postfix syntax.
Not that it matters, but it is probably a little more efficient to make the regex a regex above with `.r` and then match with that pattern here.

dilipbiswal replied:
@srowen Thanks.. Sure... I will make the change. I wanted to confirm one thing.. so initially i had it as a regex as i thought, it may be easier for us to add extensions other than `.sql` in the future. But if we are going to stay with `.sql`, should we just do a `endsWith` check instead ? WDYT ?

srowen replied:
All sounds fine. The regex could match more patterns later, or could be many regexes. Or could indeed just check for suffixes for now for simplicity and make it more complicated later.

SparkQA commented:
**[Test build #104492 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104492/testReport)** for PR 24333 at commit [`4c0f1bd`](https://github.com/apache/spark/commit/4c0f1bd61339188755a90d517cea143d67564810).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/104492/
Test PASSed.

SparkQA commented:
**[Test build #104493 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104493/testReport)** for PR 24333 at commit [`eea0a25`](https://github.com/apache/spark/commit/eea0a250f20edb0faadd7ae9c0ee8fcf6098e8d1).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/104493/
Test PASSed.

dilipbiswal committed:
fix

SparkQA commented:
**[Test build #104494 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104494/testReport)** for PR 24333 at commit [`78442e3`](https://github.com/apache/spark/commit/78442e3dea8d32e93ad6b25f79df0f66f1a1e7f1).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/104494/
Test PASSed.

dongjoon-hyun reviewed (commented):
Please update the PR title since this becomes a general approach. (Not specifically for `.swp` and `.swo`)

dilipbiswal committed:
endsWith

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9846/
Test PASSed.

SparkQA commented:
**[Test build #104519 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104519/testReport)** for PR 24333 at commit [`7600b38`](https://github.com/apache/spark/commit/7600b38bfabeef8e53d4a068e042fceeef54c815).

dilipbiswal commented:
@dongjoon-hyun Thanks.. I have updated the title and description. Did we need a JIRA for this change ? Initially since i just added couple of cases to ignore list, i hadn't created a JIRA.

dongjoon-hyun commented:
I updated the PR title, @dilipbiswal . Please create a JIRA with that title, too. Thanks!

attilapiros reviewed (commented):
pending tests otherwise LGTM

SparkQA commented:
**[Test build #104519 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104519/testReport)** for PR 24333 at commit [`7600b38`](https://github.com/apache/spark/commit/7600b38bfabeef8e53d4a068e042fceeef54c815).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/104519/
Test PASSed.

dongjoon-hyun reviewed (approved):
+1, LGTM. Merged to master.
Thank you, @dilipbiswal , @srowen , @attilapiros , @HyukjinKwon !

dilipbiswal commented:
Thanks a lot @dongjoon-hyun @srowen @HyukjinKwon @attilapiros
